pkg/xds/context/context.go

```diff
@@ -74,3 +74,41 @@ func (mc *MeshContext) GetLoggingBackend(tl *core_mesh.TrafficLogResource) *mesh
 		return lb
 	}
 }
+
+// MeshContexts is an aggregate of all MeshContexts across all meshes
+type MeshContexts struct {
+	Hash               string
+	Meshes             []*core_mesh.MeshResource
+	MeshContextsByName map[string]MeshContext
+	ZoneEgressByName   map[string]*core_mesh.ZoneEgressResource
+}
+
+func (m MeshContexts) AllDataplanes() []*core_mesh.DataplaneResource {
+	var resources []*core_mesh.DataplaneResource
+	for _, meshCtx := range m.MeshContextsByName {
```

Comment:
Each individual MeshContext has all resources sorted. I think we should keep the order for aggregated meshContexts, so generator could rely on this.